### ISSUE TITLE
[gmp] Build error

### ISSUE THREAD

**Issue by wuwenhai0** [microsoft/vcpkg#32470]
Package: gmp:x64-windows -> 6.2.1#20

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.36.32535.0
-    vcpkg-tool version: 2023-06-22-f19f3d9939100085fe2f7e41502d60b08e2083b6
    vcpkg-scripts version: dafef74af 2023-07-07 (3 days ago)

**To Reproduce**

`vcpkg install gmp:x64-windows`

**Failure logs**

```
-- Using cached gmp-6.2.1.tar.xz.
-- Extracting source D:/lib/vcpkg/downloads/gmp-6.2.1.tar.xz
-- Applying patch asmflags.patch
-- Applying patch cross-tools.patch
-- Applying patch subdirs.patch
-- Applying patch msvc_symbol.patch
-- Applying patch arm64-coff.patch
-- Applying patch gmp-arm64-asm-fix-5f32dbc41afc.patch
CMake Error at scripts/cmake/z_vcpkg_apply_patches.cmake:34 (message):
  Applying patch failed: Checking patch mpn/arm64/aors_n.asm...

  error: while searching for:

  EPILOGUE()

  PROLOGUE(func_n)

  	CLRCY

  L(ent):	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x7, [up]



  error: patch failed: mpn/arm64/aors_n.asm:68

  error: mpn/arm64/aors_n.asm: patch does not apply

  Checking patch mpn/arm64/aorsmul_1.asm...

  error: while searching for:



  include(`../config.m4')



  C	 cycles/limb

  C Cortex-A53	9.3-9.8

  C Cortex-A57	 7.0

  C X-Gene	 5.0



  C NOTES

  C * It is possible to keep the carry chain alive between the addition
  blocks



  error: patch failed: mpn/arm64/aorsmul_1.asm:32

  error: mpn/arm64/aorsmul_1.asm: patch does not apply

  Checking patch mpn/arm64/aorsorrlshC_n.asm...

  error: while searching for:



  ASM_START()

  PROLOGUE(func_n)

  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x5, [up]

  	tbnz	n, #1, L(b11)



  L(b01):	ldr	x11, [vp]

  	cbz	x18, L(1)

  	ldp	x8, x9, [vp,#8]

  	lsl	x13, x11, #LSH

  	ADDSUB(	x15, x13, x5)



  error: patch failed: mpn/arm64/aorsorrlshC_n.asm:65

  error: mpn/arm64/aorsorrlshC_n.asm: patch does not apply

  Checking patch mpn/arm64/cnd_aors_n.asm...

  error: while searching for:



  	CLRCY



  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x13, [vp]



  error: patch failed: mpn/arm64/cnd_aors_n.asm:65

  error: mpn/arm64/cnd_aors_n.asm: patch does not apply

  Checking patch mpn/arm64/logops_n.asm...

  error: while searching for:



  ASM_START()

  PROLOGUE(func)

  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x7, [up]



  error: patch failed: mpn/arm64/logops_n.asm:78

  error: mpn/arm64/logops_n.asm: patch does not apply

  Checking patch mpn/arm64/lshift.asm...

  error: while searching for:

  	add	rp, rp_arg, n, lsl #3

  	add	up, up, n, lsl #3

  	sub	tnc, xzr, cnt

  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x4, [up,#-8]



  error: patch failed: mpn/arm64/lshift.asm:61

  error: mpn/arm64/lshift.asm: patch does not apply

  Checking patch mpn/arm64/lshiftc.asm...

  error: while searching for:

  	add	rp, rp_arg, n, lsl #3

  	add	up, up, n, lsl #3

  	sub	tnc, xzr, cnt

  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x4, [up,#-8]



  error: patch failed: mpn/arm64/lshiftc.asm:61

  error: mpn/arm64/lshiftc.asm: patch does not apply

  Checking patch mpn/arm64/mul_1.asm...

  error: while searching for:



  PROLOGUE(mpn_mul_1)

  	adds	x4, xzr, xzr		C clear register and cy flag

  L(com):	lsr	x18, n, #2

  	tbnz	n, #0, L(bx1)



  L(bx0):	mov	x11, x4



  error: patch failed: mpn/arm64/mul_1.asm:56

  error: mpn/arm64/mul_1.asm: patch does not apply

  Checking patch mpn/arm64/rsh1aors_n.asm...

  error: while searching for:



  ASM_START()

  PROLOGUE(func_n)

  	lsr	x18, n, #2



  	tbz	n, #0, L(bx0)





  error: patch failed: mpn/arm64/rsh1aors_n.asm:59

  error: mpn/arm64/rsh1aors_n.asm: patch does not apply

  Checking patch mpn/arm64/rshift.asm...

  error: while searching for:

  PROLOGUE(mpn_rshift)

  	mov	rp, rp_arg

  	sub	tnc, xzr, cnt

  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x5, [up]



  error: patch failed: mpn/arm64/rshift.asm:60

  error: mpn/arm64/rshift.asm: patch does not apply

  Checking patch mpn/arm64/sqr_diag_addlsh1.asm...

  error: while searching for:

  ASM_START()

  PROLOGUE(mpn_sqr_diag_addlsh1)

  	ldr	x15, [up],#8

  	lsr	x18, n, #1

  	tbz	n, #0, L(bx0)



  L(bx1):	adds	x7, xzr, xzr



  error: patch failed: mpn/arm64/sqr_diag_addlsh1.asm:47

  error: mpn/arm64/sqr_diag_addlsh1.asm: patch does not apply

Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive.cmake:153 (z_vcpkg_apply_patches)
  ports/gmp/portfile.cmake:17 (vcpkg_extract_source_archive)
  scripts/ports.cmake:147 (include)



```

**Additional context**

**Comment by FrankXie05** [comment]
@wuwenhai0 Sorry, I didn't reproduce your problem. The patch `gmp-arm64-asm-fix-5f32dbc41afc.patch` is ok.
Could you please clear the cache and try again? :)
```
PS E:\update\vcpkg> ./vcpkg install gmp:x64-windows
Computing installation plan...
The following packages will be built and installed:
    gmp[core]:x64-windows -> 6.2.1#20
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from E:\Binarycache in 306 us. Use --debug to see more details.
Installing 1/1 gmp:x64-windows...
Building gmp[core]:x64-windows...
-- Downloading https://ftpmirror.gnu.org/gmp/gmp-6.2.1.tar.xz;https://ftp.gnu.org/gnu/gmp/gmp-6.2.1.tar.xz;https://gmplib.org/download/gmp/gmp-6.2.1.tar.xz -> gmp-6.2.1.tar.xz...
-- Extracting source E:/update/vcpkg/downloads/gmp-6.2.1.tar.xz
-- Applying patch asmflags.patch
-- Applying patch cross-tools.patch
-- Applying patch subdirs.patch
-- Applying patch msvc_symbol.patch
-- Applying patch arm64-coff.patch
-- Applying patch gmp-arm64-asm-fix-5f32dbc41afc.patch
-- Using source at E:/update/vcpkg/buildtrees/gmp/src/v6.2.1-0c723d4b6f.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x64-windows
-- Getting CMake variables for x64-windows
```

**Comment by wuwenhai0** [comment]
@FrankXie05  I reinstalled vcgpk, modified the windows display language from Chinese to English, and I also modified the language of Visual Studio to English, but it still failed. This is my detailed configuration environment:

**ENV**
- Edition	Windows 10 Pro Version	22H2 Installed on	‎7/‎2/‎2023 OS build	19045.3086 Experience	Windows Feature Experience Pack 1000.19041.1000.0 
- cmake version 3.27.0-rc4
- git version 2.41.0.windows.1
- Microsoft Visual Studio Community 2022 (64-bit) - Current Version 17.6.4

**Contents of the new log file**

Package: gmp:x64-windows -> 6.2.1#20

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.36.32535.0
-    vcpkg-tool version: 2023-06-22-f19f3d9939100085fe2f7e41502d60b08e2083b6
    vcpkg-scripts version: 79a6c0949 2023-07-10 (2 hours ago)

**To Reproduce**

`vcpkg install gmp:x64-windows`

**Failure logs**

```
-- Downloading https://ftpmirror.gnu.org/gmp/gmp-6.2.1.tar.xz;https://ftp.gnu.org/gnu/gmp/gmp-6.2.1.tar.xz;https://gmplib.org/download/gmp/gmp-6.2.1.tar.xz -> gmp-6.2.1.tar.xz...
-- Extracting source D:/lib/vcpkg/downloads/gmp-6.2.1.tar.xz
-- Applying patch asmflags.patch
-- Applying patch cross-tools.patch
-- Applying patch subdirs.patch
-- Applying patch msvc_symbol.patch
-- Applying patch arm64-coff.patch
-- Applying patch gmp-arm64-asm-fix-5f32dbc41afc.patch
CMake Error at scripts/cmake/z_vcpkg_apply_patches.cmake:34 (message):
  Applying patch failed: Checking patch mpn/arm64/aors_n.asm...

  error: while searching for:

  EPILOGUE()

  PROLOGUE(func_n)

  	CLRCY

  L(ent):	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x7, [up]



  error: patch failed: mpn/arm64/aors_n.asm:68

  error: mpn/arm64/aors_n.asm: patch does not apply

  Checking patch mpn/arm64/aorsmul_1.asm...

  error: while searching for:



  include(`../config.m4')



  C	 cycles/limb

  C Cortex-A53	9.3-9.8

  C Cortex-A57	 7.0

  C X-Gene	 5.0



  C NOTES

  C * It is possible to keep the carry chain alive between the addition
  blocks



  error: patch failed: mpn/arm64/aorsmul_1.asm:32

  error: mpn/arm64/aorsmul_1.asm: patch does not apply

  Checking patch mpn/arm64/aorsorrlshC_n.asm...

  error: while searching for:



  ASM_START()

  PROLOGUE(func_n)

  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x5, [up]

  	tbnz	n, #1, L(b11)



  L(b01):	ldr	x11, [vp]

  	cbz	x18, L(1)

  	ldp	x8, x9, [vp,#8]

  	lsl	x13, x11, #LSH

  	ADDSUB(	x15, x13, x5)



  error: patch failed: mpn/arm64/aorsorrlshC_n.asm:65

  error: mpn/arm64/aorsorrlshC_n.asm: patch does not apply

  Checking patch mpn/arm64/cnd_aors_n.asm...

  error: while searching for:



  	CLRCY



  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x13, [vp]



  error: patch failed: mpn/arm64/cnd_aors_n.asm:65

  error: mpn/arm64/cnd_aors_n.asm: patch does not apply

  Checking patch mpn/arm64/logops_n.asm...

  error: while searching for:



  ASM_START()

  PROLOGUE(func)

  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x7, [up]



  error: patch failed: mpn/arm64/logops_n.asm:78

  error: mpn/arm64/logops_n.asm: patch does not apply

  Checking patch mpn/arm64/lshift.asm...

  error: while searching for:

  	add	rp, rp_arg, n, lsl #3

  	add	up, up, n, lsl #3

  	sub	tnc, xzr, cnt

  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x4, [up,#-8]



  error: patch failed: mpn/arm64/lshift.asm:61

  error: mpn/arm64/lshift.asm: patch does not apply

  Checking patch mpn/arm64/lshiftc.asm...

  error: while searching for:

  	add	rp, rp_arg, n, lsl #3

  	add	up, up, n, lsl #3

  	sub	tnc, xzr, cnt

  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x4, [up,#-8]



  error: patch failed: mpn/arm64/lshiftc.asm:61

  error: mpn/arm64/lshiftc.asm: patch does not apply

  Checking patch mpn/arm64/mul_1.asm...

  error: while searching for:



  PROLOGUE(mpn_mul_1)

  	adds	x4, xzr, xzr		C clear register and cy flag

  L(com):	lsr	x18, n, #2

  	tbnz	n, #0, L(bx1)



  L(bx0):	mov	x11, x4



  error: patch failed: mpn/arm64/mul_1.asm:56

  error: mpn/arm64/mul_1.asm: patch does not apply

  Checking patch mpn/arm64/rsh1aors_n.asm...

  error: while searching for:



  ASM_START()

  PROLOGUE(func_n)

  	lsr	x18, n, #2



  	tbz	n, #0, L(bx0)





  error: patch failed: mpn/arm64/rsh1aors_n.asm:59

  error: mpn/arm64/rsh1aors_n.asm: patch does not apply

  Checking patch mpn/arm64/rshift.asm...

  error: while searching for:

  PROLOGUE(mpn_rshift)

  	mov	rp, rp_arg

  	sub	tnc, xzr, cnt

  	lsr	x18, n, #2

  	tbz	n, #0, L(bx0)



  L(bx1):	ldr	x5, [up]



  error: patch failed: mpn/arm64/rshift.asm:60

  error: mpn/arm64/rshift.asm: patch does not apply

  Checking patch mpn/arm64/sqr_diag_addlsh1.asm...

  error: while searching for:

  ASM_START()

  PROLOGUE(mpn_sqr_diag_addlsh1)

  	ldr	x15, [up],#8

  	lsr	x18, n, #1

  	tbz	n, #0, L(bx0)



  L(bx1):	adds	x7, xzr, xzr



  error: patch failed: mpn/arm64/sqr_diag_addlsh1.asm:47

  error: mpn/arm64/sqr_diag_addlsh1.asm: patch does not apply

Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive.cmake:153 (z_vcpkg_apply_patches)
  ports/gmp/portfile.cmake:17 (vcpkg_extract_source_archive)
  scripts/ports.cmake:147 (include)



```


**Additional context**

**Comment by FrankXie05** [comment]
Could you please replace your local patch file and try again? :)

[gmp-arm64-asm-fix-5f32dbc41afc.patch](https://github.com/microsoft/vcpkg/files/12010188/gmp-arm64-asm-fix-5f32dbc41afc.patch)

**Comment by wuwenhai0** [comment]
> Could you please replace your local patch file and try again? :)
> 
> [gmp-arm64-asm-fix-5f32dbc41afc.patch](https://github.com/microsoft/vcpkg/files/12010188/gmp-arm64-asm-fix-5f32dbc41afc.patch)

After I replaced the patch and clear the cache, it still failed.The log was almost the same.
